### PR TITLE
INT B-20078 #2 Linter Fix

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Production linter has stricter requirements. Updates a few tests to not need an eslint annotation.

[ESLint prefers `i += 1` to `i++`.](https://eslint.org/docs/latest/rules/no-plusplus#rule-details)
[ESLint doesn't like Promises to return anything, instead call the resolve or reject functions.](https://eslint.org/docs/latest/rules/no-promise-executor-return#rule-details)

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the
2. Login as a
3.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: src/pages/Office/ServicesCounselingQueue/ServicesCounselingQueue.test.jsx [hunk 1]
```diff
@@ -305,8 +305,8 @@ describe('ServicesCounselingQueue', () => {
     useUserQueries.mockReturnValue(serviceCounselorUser);
 
     const moves = JSON.parse(JSON.stringify(needsCounselingMoves));
-    /* eslint-disable no-plusplus */
-    for (let i = 0; i < 30; i++) {
+
+    for (let i = 0; i < 30; i += 1) {
```
Comment (antgmann) [inline src/pages/Office/ServicesCounselingQueue/ServicesCounselingQueue.test.jsx:309]: Very interesting, well done finding this https://eslint.org/docs/latest/rules/no-plusplus#rule-details

**Review by antgmann — APPROVED** [review]

File: src/pages/Office/ServicesCounselingQueue/ServicesCounselingQueue.test.jsx (hunk 1 above)
Comment (antgmann) [reply src/pages/Office/ServicesCounselingQueue/ServicesCounselingQueue.test.jsx:309]: re-requesting review due to playwright issues :(

Comment (r-mettler) [reply src/pages/Office/ServicesCounselingQueue/ServicesCounselingQueue.test.jsx:309]: boo because I prefer the i++. linter 😠

**Review by antgmann — APPROVED** [review]
Re-approved because tests passed, but there's still a lot of flakiness in the playwright tests

**Review by r-mettler — APPROVED** [review]

**Pull request merged**